### PR TITLE
[ DEVEX-250 ] - Add support for sending an SMS when app not installed

### PR DESCRIPTION
**Pull request by rubinsingh**
The goal of this PR is to add support for sending an SMS when the user does not have the app installed. For this we'll have to peak into session storage and extract the has_app variable. The requirements are laid out in: https://branch.atlassian.net/browse/DEVEX-250

The only changes are:

1. To not have the action not add default params because the sendSMS() function already does this as required.
2. make_new_link will not be displayed in the UI as an option
3. We will not be using an event to detect the has-app variable from session storage, instead we'll bake it into the sendSMS action.

### CODE REVIEW

File: src/view/resources/linkCreator.js [hunk 1]
```diff
@@ -94,12 +96,12 @@ linkCreator.getLinkTemplate = function() {
 			var key = children[0].value;
 			var val = children[2].value;
 			if (key && val && !blackListedKeysPattern.test(key)) {
-				link.data[children[0].value] = children[2].value;
+				link.data[children[0].value.trim()] = children[2].value.trim();
```
Comment (ethanneff) [inline src/view/resources/linkCreator.js:99]: do you need to trim channel and campaign fields as well?

File: src/view/actions/sendSMS.html [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		},
+		getSettings: function() {
+			return {
+				smsLink: linkCreator.getLinkTemplate(),
+			    smsPhoneNumber: document.getElementById("phoneNumber").value.trim()
+			}
+		},
+		validate: function() {
+			return linkCreator.isLinkDataValidJSON(linkCreator.getLinkTemplate()) &&
+                  document.getElementById("phoneNumber").value !== "";
```
Comment (ethanneff) [inline src/view/actions/sendSMS.html]: indention

Comment (ethanneff) [reply src/view/actions/sendSMS.html]: js in html file?

File: src/lib/helpers/getBranchBase.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ var loadScript = require('@adobe/reactor-load-script');
 		while (i < funcs.length) {
 			createCallback(branchSdk, funcs[i++]);
 		}
-		loadScript('https://cdn.branch.io/build.min.js');
+		loadScript('http://cdn.branch.io/branch-latest.min.js');
```
Comment (ethanneff) [inline src/lib/helpers/getBranchBase.js:11]: wrong

**Comment by ethanneff** [comment]
👍

File: src/view/resources/linkCreator.js (hunk 1 above)
Comment (rubinsingh) [reply src/view/resources/linkCreator.js:99]: Yes, good call!

File: src/lib/helpers/getBranchBase.js (hunk 3 above)
Comment (rubinsingh) [reply src/lib/helpers/getBranchBase.js:11]: waan

**Pull request merged**